python/example/brunel.py

```diff
@@ -21,13 +21,14 @@
 # Samples m unique values in interval [start, end) - gid.
 # We exclude gid because we don't want self-loops.
 def sample_subset(gid, start, end, m):
-    gen = numpy.random.RandomState(gid+42)
-    s = set()
-    while len(s) < m:
-        val = gen.randint(low=start,high=end)
-        if val != gid:
-            s.add(val)
-    return s
+    end = min(end, start + m)
+    idx = numpy.arange(start, end + 1)
+    if start < gid < end:
+        numpy.delete(idx, gid - start)
```

Comment:
```suggestion
        idx = numpy.delete(idx, gid - start)
```
`.delete()` is not in-place. Also, due to the `+1` above, `end` will occur in the list, right? The function description says it returns on interval `[start,end)`.

Reply:
Yes. We over-allocate one item and delete one. Makes net zero.

Reply:
Length, yes, but at the cost of returning values in interval `[start, end]`, no? Maybe it's not a problem?

Reply:
As stands this is completly busted. It'll return something in range `[start, start + m)`.